lib/internal/process/warning.js

```diff
@@ -28,11 +29,16 @@ function setupProcessWarnings() {
 
   // process.emitWarning(error)
   // process.emitWarning(str[, name][, ctor])
-  process.emitWarning = function(warning, name, ctor) {
+  // process.emitWarning(str[, name][, ctor][, warned])
+  process.emitWarning = function(warning, name, ctor, warned) {
     if (typeof name === 'function') {
       ctor = name;
       name = 'Warning';
     }
+    if (typeof arguments[arguments.length - 1] === 'boolean') {
+      warned = arguments[arguments.length - 1];
+    }
```

Comment:
Perhaps something like...

```js
warned = Boolean(arguments.length >= 4 ?
    arguments[3] :
    arguments[arguments.length - 1]);
```

This would allow any truthy value to be passed as the `warned` and would protect against devs passing in extra arguments, e.g. `emitWarning(msg, 'DeprecationWarning', undefined, true, 0)`. The additional arguments at the end would be ignored.